File: Lab7/Question8/src/Pegasus.java

public class Pegasus extends Bird {

	public Pegasus(String name) {
		super(name, 0);
	}

	@Override
	public boolean canFly() {
		return true;
	}

	@Override
	public int getNumOfEggs() {
		System.out.println("Pegasi do not lay eggs!");
		return super.getNumOfEggs();
	}

	@Override
	public int getLegs() {
		return 4;
	}
	
	public static void testPegasus() {
		Pegasus pegasus = new Pegasus("peg");
		System.out.println(pegasus.canFly() == true);
		System.out.println(pegasus.getNumOfEggs() == 0);
		System.out.println(pegasus.getLegs() == 4);
		System.out.println(pegasus.getName() == "peg");
		Flyer peg = new Pegasus("pegasus");
		System.out.println(peg.canFly() == true);
		System.out.println(peg.getName() == "pegasus");
	}
}

File: Lab7/Pegasus.kt
class Pegasus : Bird {

    constructor(name: String) : super(name, 0)

    override fun canFly(): Boolean {
        return true
    }

    override fun getNumOfEggs(): Int {
        println("Pegasi do not lay eggs!")
        return super.getNumOfEggs()
    }

    override fun getLegs(): Int {
        return 4
    }

    companion object {
        fun testPegasus() {
            val pegasus = Pegasus("peg")
            println(pegasus.canFly() == true)
            println(pegasus.getName() == "peg")
            println(pegasus.getNumOfEggs() == 0)
            println(pegasus.getLegs() == 4)
            println(pegasus.isDangerous() == false)
            val peg: Flyer = Pegasus("pegasus")
            println(peg.isDangerous() == false)
            println(peg.canFly() == true)
            println(peg.getName() == "pegasus")
        }
    }
}